compiler/optimizations/deadCodeElimination.cpp

```diff
@@ -490,13 +490,97 @@ static bool isIndexVariable(Symbol* sym, CForLoop* loop) {
   return false;
 }
 
-static  CallExpr* generateGPUCall(FnSymbol* kernel,
-                                     std::vector<VarSymbol*> actuals) {
+static Symbol* extractUpperBoundFromLoop(CForLoop* loop) {
+  if(BlockStmt* bs = toBlockStmt(loop->testBlockGet())) {
+      if(CallExpr *call = toCallExpr(bs->body.head)) {
+        if(call->isPrimitive(PRIM_LESSOREQUAL)) {
+          if(SymExpr *symExpr = toSymExpr(call->get(2))) {
+            return symExpr->symbol();
+          }
+        }
+      }
+  }
+
+  INT_FATAL("Unexpected upper bound for loop identified for extraction as GPU kernel");
+  return nullptr;
+}
+
+static Symbol* extractLowerBoundFromLoop(CForLoop* loop) {
+  if(BlockStmt* bs = toBlockStmt(loop->initBlockGet())) {
+    for_alist(entry, bs->body) {
+      if(CallExpr *call = toCallExpr(entry)) {
+        if(call->isPrimitive(PRIM_ASSIGN)) {
+          if(SymExpr *symExpr = toSymExpr(call->get(2))) {
+            return symExpr->symbol();
+          }
+        }
+      }
+    }
+  }
+
+  INT_FATAL("Unexpected lower bound for loop identified for extraction as GPU kernel");
+  return nullptr;
+}
+
+static VarSymbol* generateNewVariable(BlockStmt* insertionPoint, const char* name, Type* type) {
+  VarSymbol *var = new VarSymbol(name, type);
+  var->defPoint = new DefExpr(var);
```

Comment:
I think we also have a `new_Temp` helper that generates `DefExpr` under the hood. But just FYI, no need to change it here.

Reply:
Looks like there's a newTemp function in `symbol.h,.cpp`.  Looking at the code for that it looks like it generate a unique name and flags the variable as a temporary. It doesn't look like it generates a `DefExpr` though.

Reply:
Oh, sorry for the confusion, then.